docker-images/kafka-based/kafka/scripts/kafka_run.sh

```diff
@@ -61,11 +61,12 @@ if [ "$STRIMZI_KRAFT_ENABLED" = "true" ]; then
 
   if [ ! -f "$KRAFT_LOG_DIR/meta.properties" ]; then
     STRIMZI_CLUSTER_ID=$(cat "$KAFKA_HOME/custom-config/cluster.id")
-    echo "Formatting Kraft storage with cluster ID $STRIMZI_CLUSTER_ID"
+    METADATA_VERSION=$(cat "$KAFKA_HOME/custom-config/metadata.version")
+    echo "Formatting Kraft storage with cluster ID $STRIMZI_CLUSTER_ID and metadata version $METADATA_VERSION"
     mkdir -p "$KRAFT_LOG_DIR"
     # Using "=" to assign arguments for the Kafka storage tool to avoid issues if the generated
     # cluster ID starts with a "-". See https://issues.apache.org/jira/browse/KAFKA-15754
-    ./bin/kafka-storage.sh format -t="$STRIMZI_CLUSTER_ID" -c=/tmp/strimzi.properties
+    ./bin/kafka-storage.sh format -t="$STRIMZI_CLUSTER_ID" -r="$METADATA_VERSION" -c=/tmp/strimzi.properties
```

Comment:
When doing a migration from ZooKeeper to KRaft, the formatting doesn't happen because the cluster id is already assigned to the broker (from its previous life as ZooKeeper-based). What does it mean in terms of metadata version then? I mean, the cluster is migrated to KRaft but the storage won't be formatted passing the metadata version.

Reply:
I guess you would need to check it next time you are on a migrated cluster. My guess is that one of the following happens:
* It will be migrated to the default metadata version (i.e. Kafka 3.6.0 to 3.6-IV2) => this is what happens if you don't pass the `-r` method to the `kafka-storage.sh` script.
* It will use whatever is in the `inter.broker.protocol.version` => because this is essentially replacement for it.

I do not expect this to cause a major issue. We should just explain in the docs to the users what should they set as `.spec.kafka.metadataVersion`.

Reply:
I had a quick look at the impact of this on the migration side. I see that at the end of the migration, the metadata version is set with the default value you mentioned, and it seems to work okay.
Of course, when creating the controllers in my migration PR, the `-r="$METADATA_VERSION"` is still missing. We'll know more when this PR is merged and I can rebase.